addons/project/models/project.py

```diff
@@ -656,7 +734,9 @@ def write(self, vals):
             vals['date_assign'] = now
 
         result = super(Task, self).write(vals)
-
+        # rating on stage
+        if 'stage_id' in vals and vals.get('stage_id'):
```

Comment:
```python
if vals.get('stage_id'):
```
should be sufficient, don't you think?

Reply:
no, if you don't update the stage_id, you don't want to send mail. :)

Reply:
corner case, should be explained with a comment then ;)